File: backend/models/source_context.py
from pydantic import BaseModel
from typing import List


class ExtractedField(BaseModel):
    field_name: str
    value: str
    confidence: str  # "high" | "medium" | "low"
    source_snippet: str


class SourceExtractionResult(BaseModel):
    fields: List[ExtractedField]
    raw_text_length: int
    target_accelerator: str

File: frontend/src/types/sourceContext.ts
export interface ExtractedField {
  field_name: string;
  value: string;
  confidence: 'high' | 'medium' | 'low';
  source_snippet: string;
}

export interface SourceExtractionResult {
  fields: ExtractedField[];
  raw_text_length: number;
  target_accelerator: string;
}
